### PR TITLE
⭐ new: uTLS.wasm

### PR DESCRIPTION
**Pull request by gaukas**
This pull request introduces a new WebAssembly Transport Module: `uTLS.wasm`, which is based on [refraction-networking/utls](https://github.com/refraction-networking/utls) and built from a slightly modified fork.

### CODE REVIEW

**Comment by erikziyunchi** [comment]
LGTM! Thanks a lot for introducing such exciting example!!! This will be the game killer especially if we can get WATER-rs run it successfully! BTW, for the code review, would you mind pointing out which part of the `x/sys` repo have you patched?

File: x/sys/cpu/runtime_auxv_go121.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// Copyright 2023 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build go1.21 && !(wasi || wasm || wasip1 || wasip2)
```
Comment (gaukas) [inline x/sys/cpu/runtime_auxv_go121.go:5]: Originally
```go
//go:build go1.21
```

File: x/sys/cpu/runtime_auxv_go121_test.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
+// Copyright 2023 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build go1.21 && !(wasi || wasm || wasip1 || wasip2)
```
Comment (gaukas) [inline x/sys/cpu/runtime_auxv_go121_test.go:5]: Originally
```go
//go:build go1.21
```

**Comment by gaukas** [comment]
> would you mind pointing out which part of the x/sys repo have you patched?

Sure. See the review comment.

**Comment by gaukas** [comment]
> if we can get WATER-rs run it successfully

We should be if WATER-rs allows importing a Dialer-only WATM. We can create PHONY Listener and Relay for it but I think in the end both WATER runtime needs to support WATMs implementing a specific role instead of all roles. It doesn't make much sense to create a **TLS Client Listener** anyways.

**Comment by erikziyunchi** [comment]
> We should be if WATER-rs allows running a Dialer-only WATM.

Yeah, it actually does! I can try to give it a test later, so excited! And I totally agree with this idea.

**Review by erikziyunchi — APPROVED** [review]
My browser finally letting me to approve! LGTM, thanks a lot for adding the `utls.wasm`!

**Pull request merged**